.github/workflows/tests.yml

```diff
@@ -65,39 +65,96 @@ jobs:
         # update the branch protection rules on the Flowzone repository settings
         repository:
           [
+            balena-io-modules/open-balena-base,
             balena-io/balena-api,
             balena-io/docs,
             balena-io/environment-staging,
-            balena-io-modules/open-balena-base,
             balena-os/balena-engine,
+            balena-os/fatrw,
             balenaltd/handbook,
             product-os/environment-staging,
-            balena-os/fatrw,
           ]
+        include:
+          - token_scope: >-
+              {
+                "actions": "read",
+                "administration": "write",
+                "checks": "read",
+                "contents": "write",
+                "members": "read",
+                "metadata": "read",
+                "pull_requests": "write",
+                "statuses": "read",
+                "workflows": "write"
+              }
+          - app_id: "${{ vars.APP_ID || 291899 }}"
+          - installation_id: 34046903
+            repository: balena-io-modules/open-balena-base
+          - installation_id: 34046749
+            repository: balena-io/balena-api
+          - installation_id: 34046749
+            repository: balena-io/docs
+          - installation_id: 34046749
+            repository: balena-io/environment-staging
+          - installation_id: 34046907
+            repository: balena-os/balena-engine
+          - installation_id: 34046907
+            repository: balena-os/fatrw
+          - installation_id: 34046911
+            repository: balenaltd/handbook
+          - installation_id: 34040165
+            repository: product-os/environment-staging
 
     env:
       # environment variables used by gh CLI
       # https://cli.github.com/manual/gh_help_environment
       GH_REPO: ${{ matrix.repository }}
-      GH_TOKEN: ${{ secrets.FLOWZONE_TOKEN }}
       GH_PROMPT_DISABLED: "true"
       GH_DEBUG: "true"
       GH_PAGER: "cat"
       pr_branch: "flowzone/${{ github.head_ref || github.ref_name }}"
       pr_title: "Test Flowzone PR ${{ github.event.pull_request.number }}"
 
     steps:
+      - name: Check secrets
+        id: private_key
+        run: |
+          if [[ -z '${{ secrets.FLOWZONE_TOKEN }}' ]] \
+            && [[ -z '${{ secrets.PRIVATE_KEY }}' ]]; then
+              echo '::error::Must specify either PRIVATE_KEY or FLOWZONE_TOKEN.'
+              false
+          fi
+
+          if [[ -n '${{ secrets.PRIVATE_KEY }}' ]]; then
+              echo 'available=true' >> $GITHUB_OUTPUT
+          else
+              echo 'available=false' >> $GITHUB_OUTPUT
+          fi
+
+      - name: Generate token
+        uses: tibdex/github-app-token@v1
```

Comment:
We should pin this to a full semver since it's not an official github action.

Once this merges Renovate will pin it to a commit hash, and keep the version as a comment for tracking.